src/mixins/yield-erc/ERC20Vault.sol

```diff
@@ -0,0 +1,149 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+pragma solidity 0.8.10;
+
+import {ERC20} from "../../tokens/ERC20.sol";
+import {FixedPointMathLib} from "../../utils/FixedPointMathLib.sol";
+import {SafeTransferLib} from "../../utils/SafeTransferLib.sol";
+
+/// @title Yield Bearing Vault
+/// @author joeysantoro, Transmissions11 and JetJadeja
+contract ERC4626Vault is ERC20 {
+    using SafeTransferLib for ERC20;
+    using FixedPointMathLib for uint256;
+
+    /*///////////////////////////////////////////////////////////////
+                                IMMUTABLES
+    //////////////////////////////////////////////////////////////*/
+
+    /// @notice The underlying token the Vault accepts.
+    ERC20 public immutable underlying;
+
+    /// @notice The base unit of the underlying token and hence vault.
+    /// @dev Equal to 10 ** decimals. Used for fixed point arithmetic.
+    uint256 public immutable baseUnit;
```

Comment:
this should be internal now that it is no longer enshrined in the EIP

Reply:
maybe we make it all caps again as well

Reply:
it's true that it's not a true `constant` but it can be treated as a variant of a constant because after `constructor` act as it. 

For that reason (but it's more a stylistic way to instantly recognize it) it should be uppercase. but this go in the bucket of style choice.